/*
 * Copyright 2015 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// Author: jmaessen@google.com (Jan-Willem Maessen)

syntax = "proto2";

package net_instaweb;

message MobilizeMenu {
  repeated MobilizeMenuItem entries = 1;
}

message MobilizeMenuItem {
  // A menu entry ought to have a name and one of a link url or a submenu.
  // Note that (for the moment) there should be no HTML markup here.  Markup
  // will require explict DOM (or breaking our parsing model by injecting
  // strings with markup that aren't seen by our filters).
  optional string name = 1 [ default = "" ];
  optional string url = 2 [ default = "" ];
  optional MobilizeMenu submenu = 3;
}
